internal/provider/azure/internal/errorutils/errors.go

```diff
@@ -215,24 +212,13 @@ func SimpleError(err error) error {
 // HandleCredentialError determines if given error relates to invalid credential.
 // If it is, the credential is invalidated.
 // Original error is returned untouched.
-func HandleCredentialError(err error, ctx envcontext.ProviderCallContext) error {
-	MaybeInvalidateCredential(err, ctx)
-	return err
-}
-
-// MaybeInvalidateCredential determines if given error is related to authentication/authorisation failures.
-// If an error is related to an invalid credential, then this call will try to invalidate that credential as well.
-func MaybeInvalidateCredential(err error, ctx envcontext.ProviderCallContext) bool {
-	if !HasDenialStatusCode(err) {
-		return false
+func HandleCredentialError(ctx context.Context, invalidator environs.CredentialInvalidator, err error) (bool, error) {
+	if err == nil {
+		return false, nil
 	}
 
 	converted := fmt.Errorf("azure cloud denied access: %w", common.CredentialNotValidError(err))
```

Comment:
I see why you did it, but I think it was easier to always return the error from the call sites and leave `common.HandleCredentialError()` to actually check the error and return a bool.

Reply:
I'm torn, let me think about this one. It's so they could do this, but the returning of the bool is a bit annoying. I'll make a function that discards the bool, so they can do that naturally.

```
if err != nil {
    return errors.Trace(HandleCredentialError(ctx, invalidator))
}
```

Reply:
This is much nicer now.